cmd/pipeline-template/plan_test.go

```diff
@@ -0,0 +1,251 @@
+	}
+	defer os.Remove(tempFile.Name())
+
+	args := []string{"pipeline-template", "plan", "--config", tempFile.Name(), "--gate-endpoint", ts.URL}
+	currentCmd := NewPlanCmd(pipelineTemplateOptions{})
+	rootCmd := getRootCmdForTest()
+	pipelineTemplateCmd := NewPipelineTemplateCmd(os.Stdout)
+	pipelineTemplateCmd.AddCommand(currentCmd)
+	rootCmd.AddCommand(pipelineTemplateCmd)
+
+	rootCmd.SetArgs(args)
+	err := rootCmd.Execute()
+	if err == nil {
+		t.Fatalf("Expected failure but command succeeded")
+	}
+}
+
+func TestPipelineTemplatePlan_flags(t *testing.T) {
+	ts := gateServerPlanSuccess()
+	defer ts.Close()
+
+	args := []string{"pipeline-template", "plan", "--gate-endpoint", ts.URL} // Missing pipeline config file and stdin.
+	currentCmd := NewPlanCmd(pipelineTemplateOptions{})
+	rootCmd := getRootCmdForTest()
+	pipelineTemplateCmd := NewPipelineTemplateCmd(os.Stdout)
+	pipelineTemplateCmd.AddCommand(currentCmd)
+	rootCmd.AddCommand(pipelineTemplateCmd)
+
+	rootCmd.SetArgs(args)
+	err := rootCmd.Execute()
+	if err == nil {
+		t.Fatalf("Expected failure but command succeeded")
+	}
+}
+
+// gateServerUpdateSuccess spins up a local http server that we will configure the GateClient
+// to direct requests to. Responds with 404 NotFound to indicate a pipeline template doesn't exist,
+// and Accepts POST calls.
+func gateServerPlanSuccess() *httptest.Server {
+	mux := http.NewServeMux()
+	mux.Handle("/v2/pipelineTemplates/plan", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodPost {
+			fmt.Fprintln(w, strings.TrimSpace(testPipelineTemplatePlanResp))
+		} else {
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+	return httptest.NewServer(mux)
+}
+
+const testPipelineTemplatePlanResp = `
+{
+ "application": "app",
+ "description": "",
+ "expectedArtifacts": [],
+ "id": "unknown",
+ "keepWaitingPipelines": false,
+ "lastModifiedBy": "anonymous",
+ "limitConcurrent": true,
+ "name": "My First SpEL Pipeline",
+ "notifications": [],
+ "parameterConfig": [],
+ "stages": [
+  {
+   "inject": {
+    "first": true,
+    "last": false
+   },
+   "notifications": [],
+   "refId": "wait0",
+   "requisiteStageRefIds": [],
+   "type": "wait",
+   "waitTime": 2
+  },
+  {
+   "name": "My Wait Stage",
+   "notifications": [],
+   "refId": "wait1",
+   "requisiteStageRefIds": [
+    "wait0"
+   ],
+   "type": "wait",
+   "waitTime": 6
+  },
+  {
+   "notifications": [],
+   "refId": "wait2",
+   "requisiteStageRefIds": [
+    "wait1"
+   ],
+   "type": "wait",
+   "waitTime": 67
+  }
+ ],
+ "templateVariables": {
+  "waitTime": 6
+ },
+ "trigger": {
+  "artifacts": [],
+  "parameters": {},
+  "type": "manual",
+  "user": "anonymous"
+ },
+ "triggers": [
+  {
+   "attributeConstraints": {},
+   "enabled": true,
+   "payloadConstraints": {},
+   "pubsubSystem": "google",
+   "source": "jack",
+   "subscription": "super-derp",
+   "subscriptionName": "super-derp",
+   "type": "pubsub"
+  }
+ ],
```

Comment:
The json tabbing between the test data is different -- intentional?

Reply:
I passed one through `json_pp` and not the other, make them consistent.